### PR TITLE
chore!: remove amino support for x/mint

### LINKED ISSUE #2069: Remove support for legacy `amino` encoding in the `mint` module
## Context

https://github.com/celestiaorg/celestia-app/blob/60f932802e30c31c2f9dfa810bce0e74902cf716/x/mint/keeper/querier.go#L14-L31

and

https://github.com/celestiaorg/celestia-app/blob/60f932802e30c31c2f9dfa810bce0e74902cf716/x/mint/types/codec.go#L8-L13

## Problem

Celestia supports new versions of the cosmos-sdk that use protobuf for encoding, and deprecate `amino`. So we can remove support for amino for the `mint` module.

## Proposal

Remove `amino` related implementation so we don't have to maintain all that code.

### PR DESCRIPTION
**Pull request by NishantKoyalwar**
## Overview

- support removed for legacy amino encoding in the mint module
-  closes #2069 

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: x/mint/module.go [hunk 1]
```diff
@@ -110,8 +110,8 @@ func (AppModule) QuerierRoute() string {
 }
 
 // LegacyQuerierHandler returns the mint module sdk.Querier.
```
Comment (rootulp) [inline x/mint/module.go:112]:
```suggestion
// LegacyQuerierHandler returns nil because this mint module doesn't support the
// Amino codec.
```

**Comment by rootulp** [comment]
@NishantKoyalwar I updated the PR description to conform to conventional commits. See [contributing](https://github.com/celestiaorg/celestia-app#contributing) for more details.

**Review by rootulp — CHANGES_REQUESTED** [review]
Is it also safe to delete https://github.com/celestiaorg/celestia-app/blob/60f932802e30c31c2f9dfa810bce0e74902cf716/x/mint/types/codec.go ?

**Review by SweeXordious — COMMENTED** [review]
yes, I guess it's safe to delete the codec as well

**Review by cmwaters — APPROVED** [review]

File: x/mint/keeper/genesis.go [hunk 2]
```diff
@@ -6,8 +6,7 @@ import (
 )
 
 // InitGenesis initializes the x/mint store with data from the genesis state.
-func (k Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper, data *types.GenesisState) {
-	k.SetMinter(ctx, data.Minter)
+func (k Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper) {
```
Comment (SweeXordious) [inline x/mint/keeper/genesis.go:9]: I think we still need to set the minter struct here because it's used directly in the `BeginBlock`

Comment (rootulp) [reply x/mint/keeper/genesis.go:9]: +1 tests on this PR fail: https://github.com/celestiaorg/celestia-app/actions/runs/5522450575/jobs/10072054294?pr=2073#step:4:590

File: x/mint/keeper/genesis.go [hunk 3]
```diff
@@ -7,8 +7,8 @@ import (
 
 // InitGenesis initializes the x/mint store with data from the genesis state.
 func (k Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper, data *types.GenesisState) {
-	k.SetMinter(ctx, data.Minter)
 	// override the genesis time with the actual genesis time supplied in `InitChain`
+	k.SetMinter(ctx, data.GetMinter())
```
Comment (cmwaters) [inline x/mint/keeper/genesis.go:11]: I don't think this change does anything. Can we just have it as before to keep the PR diff low

File: go.mod [hunk 4]
```diff
@@ -45,9 +45,10 @@ require (
 	github.com/docker/go-units v0.5.0 // indirect
 	github.com/opencontainers/go-digest v1.0.0 // indirect
 	github.com/opencontainers/image-spec v1.1.0-rc2 // indirect
-	golang.org/x/mod v0.9.0 // indirect
-	golang.org/x/tools v0.7.0 // indirect
+	golang.org/x/mod v0.10.0 // indirect
+	golang.org/x/tools v0.8.0 // indirect
 	gotest.tools/v3 v3.4.0 // indirect
+	mvdan.cc/gofumpt v0.5.0 // indirect
```
Comment (cmwaters) [inline go.mod:51]: Is there any reason why these dependencies were bumped? Can this be done in a separate PR

**Comment by rootulp** [comment]
Sorry for the tedious rounds of review on this @NishantKoyalwar . I'll close this PR for now but feel free to re-open if you'd like to continue working on it.
